src/emulator/controller.ts

```diff
@@ -421,6 +422,41 @@ export async function startAll(options: EmulatorOptions, showUI: boolean = true)
       options.config.src.functions.source
     );
 
+    emulatableBackends.push({
+      functionsDir,
+      env: {
+        ...options.extensionEnv,
```

Comment:
can you document why we need extensionEnv when starting functions emu?

also wondering if we can do better than this? perhaps something like 
```
...(shouldStart(options, Emulators.EXTENSIONS) ? options.extensionEnv : [])
```

Reply:
Will do. extensionsEnv is a (now very confusingly named) remnant from the local Extensions emulator commands (firebase ext:dev:emulators:*) - it is unused by the Extensions Emulator. Eventually, I want to get rid of those commands entirely and have publishers emulate the extensions they are testing by pointing an extension manifest at the local directory.

I'll go thru and rename  + comment the parts that are there only to support firebase ext:dev:emulators:*. Does extDevEnv sound like a reasonable name?

Reply:
I see thanks a lot for the explanation! yes that sounds like a good name :)

Reply:
https://github.com/firebase/firebase-tools/pull/4150 Put these changes in a separate PR so its easy to see - i'm going to merge that into this then merge this.